custom_completions_test.go

```diff
@@ -1194,6 +1195,44 @@ func TestFlagDirFilterCompletionInGo(t *testing.T) {
 	}
 }
 
+func TestValidArgsFuncCmdContext(t *testing.T) {
```

Comment:
There are two reasons for this problem:
1. the code in the `validArgsFunc` must extract the context using `cmd.Context()`
2. `executeCommandWithContext` should be called on a sub-command.  Calling it on the root command is not a sufficient test because the root command already has the context set.

Reply:
Thanks, good catch!

I made sure 1. does not happen again. I also call completion on a child command now. removing `finalCmd.ctx = c.ctx` now correctly fails the test.